shared/lib/transactions-controller-utils.js

```diff
@@ -35,6 +35,52 @@ export function calcTokenAmount(value, decimals) {
   return new BigNumber(String(value)).div(multiplier);
 }
 
+export function stripAddressFromLogTopic(addressLogTopic) {
+  return `0x${addressLogTopic.replace(/^0x000000000000000000000000/u, '')}`;
+}
+
+/**
+ * Extracts token transfers from a txReceipt by filtering through logs.
+ *
+ * @param txReceipt - A transaction receipt.
+ * @returns {{ contractAddress: string, to: string, from: string }} A list of token transfers from a txReceipt.
+ */
+export function getTokenTransfersFromTxReceipt(txReceipt) {
+  const txReceiptLogs = txReceipt?.logs;
+  if (!txReceiptLogs || txReceipt?.status === '0x0') {
+    return [];
+  }
+  return txReceiptLogs
+    .map((txReceiptLog) => {
+      if (!txReceiptLog.topics) {
+        return undefined;
+      }
+      const isTokenTransfer =
+        txReceiptLog.topics[0] &&
+        txReceiptLog.topics[0] === TOKEN_TRANSFER_LOG_TOPIC_HASH;
+      if (!isTokenTransfer) {
+        return undefined;
+      }
+
+      const transferFromAddressTopic = txReceiptLog.topics[1];
+      if (!transferFromAddressTopic) {
+        return undefined;
+      }
+
+      const transferToAddressTopic = txReceiptLog.topics[2];
+      if (!transferToAddressTopic) {
+        return undefined;
+      }
+
+      return {
+        to: stripAddressFromLogTopic(transferToAddressTopic),
+        from: stripAddressFromLogTopic(transferFromAddressTopic),
+        contractAddress: txReceiptLog.address,
+      };
+    })
+    .filter((transfer) => Boolean(transfer));
+}
```

Comment:
same here , you can use only reduce function with if statement instead of map then filter